src/components/course/data/hooks.js

```diff
@@ -254,3 +298,58 @@ export function useCourseEnrollmentUrl({
 
   return enrollmentUrl;
 }
+
+/**
+ * Hook function for data to extract subsidy information from CourseContext and UserSubsidyContext
+ *
+ * @param { object } location router location.
+ *   (TODO Refactor this once useEnrollmentUrl is refactored!)
+ *
+ * @returns {object} with fields:
+ * {
+ *    subscriptionLicense,
+ *    userSubsidy,
+ *    enrollmentUrl,
+ *    offersCount,
+*     courseHasOffer,
+ * }
+ */
+export function useSubsidyData({ location }) {
```

Comment:
nit: maybe `useSubsidyDataforCourse` to make it clear that this hook is course-specific. In case future us see `useSubsidyData` and think it's a more generic hook that doesn't involve a course.